notes/2024-02-06 biglep feedback for PR #65.md

```diff
@@ -0,0 +1,99 @@
+https://github.com/libp2p/github-mgmt/pull/12#pullrequestreview-999621620 speaks to how there was confusion when folks were @mentioned and that they thought they were being removed from the org.
+
+Even if we don't give the full user-oriented view of one's permissions, it would be great if we gave a user-oriented summary of the changes.  Example:
+
+```
+@biglep
+Removed from repos: repoName1/permissoinLevel1, repoName2/permissoinLevel2
+Removed from teams: team1
+```
+
+That by itself will cut down on some of the confusion, but it will still need a disclaimer (e.g., "Even though your direct repo permissions have been removed, you may still have access through a team.  Please check the full diff.").
+
+## More intelligent sorting of keys
+I wish github-mgmt didn’t sort all key alphabetically when there is a more meaningful default.  
+For example, with "collaborators” and “teams”, I want the keys be in descencing order: admin > maintain > push > triage > pull.
+Similarly, I want crucial properties like "archived" to be at the top.
+I think the default otherwise can be alphabetical.
+(Implementation wise I assume we do a custom sort function that pulls out any specific keys, sorts by that specified order, puts those first, and then adds the rest of the keys alphabetically.)
+
+# Somewhere between general and IPLD-specific
+These are at least things that apply to the 2024Q1 cleanup for ipld, libp2p, and ipfs orgs. Some of them can be generalized.
+
+## Archived repos cause clutter
+These .yaml files can be unwieldy.  One thing I think could help would be to have a view on all the repos that aren't archived.  Some ideas (I assume from least to most amount of work):
+* Put all the archived repos at the bottom of the file.  (I still don't like this one as much because archived repos will show up when doing text search for "team X" when trying to answer "what teams does X have permissions to").
+* Have an optional separate .yaml that merges in at build/process time.  This would allow a dedicated separate file (e.g., archived_repos.yaml).
+  * Generalizing this, maybe github-mgmt can be configured to do a merge of any specified files, which lets repo organizers decide if/how they want to break up their yaml file in general?
+* Go the IPFS route and actually have an "ipfs-inactive" org.  Archived projects can get moved there.  This makes things very clear about the maintenance status and also solves the declutter problem. 
+
+## Strip out access permissions to archived repos
+It adds clutter having individuals and teams showing up with access permissions in archived repos.  
+If a repo is archived, I think we should strip permissions.  
+Someone can always unarchive and add permissions through github-mgmt.  
+In addition to this reducing clutter, I this proposed process is good because it gives clear visibility to a significant repo event (e.g., unarchiving).
+
+## Remove the w3dt-stewards team
```

Comment:
2024-02-08 conversation: agreed this should get done.

Reply:
@BigLep is this something we will execute soon, a long term plan? This group has been a "maintainer" team in places like https://github.com/ipfs/someguy/settings/access and if we just remove it, it will disrupt day-to-day

Reply:
It was clear-cut to do in ipld land.
I know more thought will be needed in ipfs. 

At the minimum what we'd do in IPFS is rename the team (e.g., ip-shipyard) and reduce the people who are on that team to those who are actually on shipyard.

Master tracking issue for when we get into ipfs org: https://github.com/ipfs/ipfs/issues/511